lib/src/rotate.dart

```diff
@@ -40,10 +40,9 @@ abstract class _Rotate extends Module {
   /// represented by the maximum value of [rotateAmount].
   _Rotate(this._direction, Logic original, Logic rotateAmount,
       {int? maxAmount, super.name = 'rotate'})
-      : maxAmount = min(
-          maxAmount ?? original.width,
-          pow(2, rotateAmount.width).toInt() - 1,
-        ) {
+      : maxAmount = min(maxAmount ?? original.width,
+            pow(2, rotateAmount.width).toInt() - 1),
+        super(definitionName: '_Rotate_${_direction}_W${original.width}') {
```

Comment:
Why not `Rotate` instead of `_Rotate`?  You probably want `_direction.name` so you don't get a big prefix of the enum name